tests/test_analyzer.py

```diff
@@ -0,0 +1,509 @@
+#
+# Copyright (c) nexB Inc. and others. All rights reserved.
+# ScanCode is a trademark of nexB Inc.
+# SPDX-License-Identifier: Apache-2.0
+# See http://www.apache.org/licenses/LICENSE-2.0 for the license text.
+# See https://github.com/nexB/scancode-toolkit for support or download.
+# See https://aboutcode.org for more information about nexB OSS projects.
+#
+
+import attr
+import os
+
+from commoncode.testcase import FileBasedTesting
+
+from results_analyze.df_file_io import DataIOJSON
+
+from results_analyze import analyzer
+
+
+class TestAnalyzer(FileBasedTesting):
+    test_data_dir = os.path.join(os.path.dirname(__file__), "data/analyzer/")
+
+    def test_analyzer_is_correct_detection_case_all_3_seq(self):
+
+        test_file = self.get_test_loc(
+            "analyzer_is_correct_detection_case_all_3_seq.json"
+        )
+        license_matches = DataIOJSON.load_json(test_file)
+
+        assert not analyzer.is_correct_detection(license_matches)
+
+    def test_analyzer_is_correct_detection_case_all_1_hash(self):
+
+        test_file = self.get_test_loc(
+            "analyzer_is_correct_detection_case_all_1_hash.json"
+        )
+        license_matches = DataIOJSON.load_json(test_file)
+
+        assert analyzer.is_correct_detection(license_matches)
+
+    def test_analyzer_is_correct_detection_case_mixed_123(self):
+
+        test_file = self.get_test_loc(
+            "analyzer_is_correct_detection_case_mixed_123.json"
+        )
+        license_matches = DataIOJSON.load_json(test_file)
+
+        assert not analyzer.is_correct_detection(license_matches)
+
+    def test_analyzer_is_correct_detection_case_mixed_14(self):
+
+        test_file = self.get_test_loc(
+            "analyzer_is_correct_detection_case_mixed_14.json"
+        )
+        license_matches = DataIOJSON.load_json(test_file)
+
+        assert analyzer.is_correct_detection(license_matches)
+
+    def test_analyzer_is_match_coverage_less_than_threshold_low(self):
```

Comment:
That's a nit for later... you could use a slightly more dense style with less empty lines, for instance:
```
    def test_analyzer_is_match_coverage_less_than_threshold_low(self):
        test_file = self.get_test_loc("analyzer_is_match_coverage_less_than_threshold_low.json")
        license_matches = DataIOJSON.load_json(test_file)
        assert analyzer.is_match_coverage_less_than_threshold(
            license_matches, threshold=analyzer.IMPERFECT_MATCH_COVERAGE_THR
        )
```

... would be as readable and go from 10 to 6 lines

Reply:
Yes! that'd better for tests, I'd follow that and make them dense. Thanks!